src/coreclr/jit/importervectorization.cpp

```diff
@@ -380,15 +380,15 @@ GenTree* Compiler::impExpandHalfConstEqualsSWAR(
         //   [ ch1 ]
         //   [value]
         //
-        return impCreateCompareInd(data, TYP_SHORT, dataOffset, cns[0], cmpMode);
+        return impCreateCompareInd(data, TYP_USHORT, dataOffset, cns[0], cmpMode);
     }
     if (len == 2)
     {
         //   [ ch1 ][ ch2 ]
         //   [   value    ]
         //
         const UINT32 value = MAKEINT32(cns[0], cns[1]);
-        return impCreateCompareInd(data, TYP_INT, dataOffset, value, cmpMode);
+        return impCreateCompareInd(data, TYP_UINT, dataOffset, value, cmpMode);
```

Comment:
Doesn't this end up creating `TYP_UINT` indirs aka invalid IR?

(Applies to a few cases below as well)

Reply:
@SingleAccretion changed to cast node